### PR TITLE
[connectors] - fix: handle invalid string ids in base resource

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at fixing a few errors caused by a wrong id that cannot be parsed as int:
```
DatabaseError: column "nan" does not exist Error at Query.run (/app/node_modules/sequelize/src/dialects/postgres/query.js:76:25) at <anonymous> (/app/node_modules/sequelize/src/sequelize.js:650:28) at process.processTicksAndRejections (node:internal/process/task_queues:95:5) at PostgresQueryInterface.select (/app/node_modules/sequelize/src/dialects/abstract/query-interface.js:1001:12) at Function.findAll (/app/node_modules/sequelize/src/model.js:1824:21) at Function.findOne (/app/node_modules/sequelize/src/model.js:1998:12) at Function.findByPk (/app/node_modules/sequelize/src/model.js:1955:12) at Function.fetchById (/app/src/resources/base_resource.ts:39:18) at _getConnector (/app/src/api/get_connector.ts:29:21) at <anonymous> (/app/src/logger/withlogging.ts:17:7)
```

**References:**
 - [Logs](https://app.datadoghq.eu/logs?query=%40error.original.message%3A%22column%20%5C%22nan%5C%22%20does%20not%20exist%22&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&event=AwAAAZNUi3kX-Eg5IwAAABhBWk5VaTNfRUFBQVVVX1FBYUc3Z0lRQVYAAAAkMDE5MzU0OTQtOTY2MC00OTliLThlMTctN2E5NzY1YjczNmY3AAAl7A&fromUser=true&link_source=monitor_notif&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1731240406636&to_ts=1732536406636&live=true)
## Risk

Low

## Deploy Plan

Deploy connectors

### CODE REVIEW

**Review by flvndvd — COMMENTED** [review]
Weird that we need this here. One of the basic rule is never trust user inputs/external inputs. Validating here seems a bit late, no?

**Comment by JulesBelveze** [comment]
Unless I'm missing something we are only checking the input type but never checking that the string can actually be converted to integer, right?
Can we add custom validation to `io-ts`?

**Comment by flvndvd** [comment]
> Unless I'm missing something we are only checking the input type but never checking that the string can actually be converted to integer, right? Can we add custom validation to `io-ts`?

Were you able to track where is it coming from? Which URL? We can indeed make io-ts stricter.

**Comment by JulesBelveze** [comment]
Here's a breakdown of the [urls](https://app.datadoghq.eu/logs?query=%40error.original.message%3A%22column%20%5C%22nan%5C%22%20does%20not%20exist%22&agg_m=count&agg_m_source=base&agg_q=%40url&agg_t=count&analyticsOptions=%5B%22bars%22%2C%22dog_classic%22%2Cnull%2Cnull%2C%22value%22%5D&cols=host%2Cservice&fromUser=true&link_source=monitor_notif&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=timeseries&from_ts=1731240406636&to_ts=1732536406636&live=true) that were hit.

**Comment by flvndvd** [comment]
> Here's a breakdown of the [urls](https://app.datadoghq.eu/logs?query=%40error.original.message%3A%22column%20%5C%22nan%5C%22%20does%20not%20exist%22&agg_m=count&agg_m_source=base&agg_q=%40url&agg_t=count&analyticsOptions=%5B%22bars%22%2C%22dog_classic%22%2Cnull%2Cnull%2C%22value%22%5D&cols=host%2Cservice&fromUser=true&link_source=monitor_notif&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=timeseries&from_ts=1731240406636&to_ts=1732536406636&live=true) that were hit.

Do we know who emit those queries? IIRC, connectors API is not directly exposed. It's probably some calls from the public api that are simply forwarded without the inputs being verified?

**Comment by JulesBelveze** [comment]
I'm not able to trace the origin of the queries. But as a first iteration I can do the check in the public API before sending the request to connectors? If it persists then we can make io-ts stricted on the connectors side. WDYT?

**Comment by flvndvd** [comment]
> I'm not able to trace the origin of the queries. But as a first iteration I can do the check in the public API before sending the request to connectors? If it persists then we can make io-ts stricted on the connectors side. WDYT?

Seems better indeed.

File: types/src/front/lib/connectors_api.ts [hunk 1]
```diff
@@ -390,6 +390,15 @@ export class ConnectorsAPI {
   async getConnector(
     connectorId: string
   ): Promise<ConnectorsAPIResponse<ConnectorType>> {
+    const parsedId = parseInt(connectorId, 10);
```
Comment (flvndvd) [inline types/src/front/lib/connectors_api.ts:393]: Just to confirm, it's supposed to always be a number here? Looking at the prototype it's confusing (`connectorId: string`).

Comment (JulesBelveze) [reply types/src/front/lib/connectors_api.ts:393]: Looks like it yes. `getConnector` is retrieving the connector this way:
```
const connector = await ConnectorResource.fetchById(req.params.connector_id);
```

Comment (JulesBelveze) [reply types/src/front/lib/connectors_api.ts:393]: I agree very confusing

**Review by flvndvd — APPROVED** [review]

File: types/src/front/lib/connectors_api.ts (hunk 1 above)
Comment (flvndvd) [reply types/src/front/lib/connectors_api.ts:393]: Make sure to monitor. You might want to log as well.

**Pull request merged**